src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,123 @@
+    private static final int INDEX_ID = 1;
+    private static final int INDEX_NAME = 2;
+    private static final int INDEX_COUNTRY = 3;
+
+    @Override
+    public Manufacturer add(Manufacturer manufacturer) {
+        if (manufacturer.getName() != null & manufacturer.getCountry() != null) {
+            try (Connection connection = ConnectionUtil.getConnection()) {
+                String insertQuery = "INSERT INTO manufacturer "
+                        + "(name,country) values(?,?)";
+                PreparedStatement createStatement =
+                        connection.prepareStatement(insertQuery,
+                                Statement.RETURN_GENERATED_KEYS);
+                createStatement.setString(INDEX_ID, manufacturer.getName());
+                createStatement.setString(INDEX_NAME, manufacturer.getCountry());
+                createStatement.executeUpdate();
+                ResultSet generatedKeys = createStatement.getGeneratedKeys();
+                if (generatedKeys.next()) {
+                    Long id = generatedKeys.getObject(1, Long.class);
+                    manufacturer.setId(id);
+                }
+            } catch (SQLException e) {
+                throw new DataProcessingException("Ca`nt create new manufacturer", e);
+            }
+        } else {
+            throw new RuntimeException("Ca`nt create manufacturer with this parameters");
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Optional<Manufacturer> get(Long id) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            String getManufacturerQuery = "SELECT * FROM manufacturers WHERE id = ?"
+                    + " AND is_deleted = FALSE;";
+            PreparedStatement preparedStatement =
+                    connection.prepareStatement(getManufacturerQuery);
+            preparedStatement.setLong(INDEX_ID, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(getManufacturerFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Ca`nt get manufacturer with this id: " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String getAllQuery = "SELECT * FROM manufacturer WHERE is_deleted = FALSE";
+        List<Manufacturer> listOfManufacture;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            listOfManufacture = new ArrayList<>();
+            Manufacturer manufacturer;
+            Statement getAllStatement = connection.createStatement();
+            ResultSet resultSet = getAllStatement.executeQuery(getAllQuery);
+            while (resultSet.next()) {
+                manufacturer = getManufacturerFromResultSet(resultSet);
+                listOfManufacture.add(manufacturer);

```

Comment:
```suggestion
                Manufacturer manufacturer = getManufacturerFromResultSet(resultSet);
                listOfManufacture.add(manufacturer);
```
I don't see any reason to move type declaration outside the loop.
Or you can just do this way:
```suggestion
               listOfManufacture.add(getManufacturerFromResultSet(resultSet));
```